src/main/java/dk/bec/polonez/reservationsystem/service/OfferService.java

```diff
@@ -2,55 +2,84 @@
+import dk.bec.polonez.reservationsystem.model.Feature;
 import dk.bec.polonez.reservationsystem.model.Offer;
-import dk.bec.polonez.reservationsystem.model.User;
+import dk.bec.polonez.reservationsystem.repository.FeatureRepository;
 import dk.bec.polonez.reservationsystem.repository.OfferRepository;
+import dk.bec.polonez.reservationsystem.repository.UserRepository;
+import org.modelmapper.ModelMapper;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.http.HttpStatus;
 import org.springframework.stereotype.Service;
-import org.springframework.web.server.ResponseStatusException;
 
 import java.util.List;
-import java.util.Optional;
+import java.util.stream.Collectors;
 
 @Service
 public class OfferService {
 
     private final OfferRepository offerRepository;
+    private final FeatureRepository featureRepository;
+    private final ModelMapper modelMapper;
 
     @Autowired
-    public OfferService(OfferRepository offerRepository) {
+    public OfferService(OfferRepository offerRepository, FeatureRepository featureRepository) {
         this.offerRepository = offerRepository;
+        this.featureRepository = featureRepository;
+        this.modelMapper = new ModelMapper();
     }
 
-    public List<Offer> getAll() {
-        return offerRepository.findAll();
+    public List<OfferDto> getAll() {
+        List<Offer> offers = offerRepository.findAll();
+        return offers.stream()
+                .map(offer -> modelMapper.map(offer, OfferDto.class))
+                .collect(Collectors.toList());
     }
 
-    public Offer getById(long id) {
-        Optional<Offer> optionalOffer = offerRepository.findById(id);
-        return optionalOffer
-                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND));
+    public OfferDto getById(long id) {
+        Offer offer = offerRepository.getById(id);
+
+        return modelMapper.map(offer, OfferDto.class);
     }
 
-    public ResponseOfferDto addOffer(CreateOfferDto offerDto) {
-        User user = new User();
-        user.setId(1L);
+    public OfferDto addOffer(CreateOfferDto offerDto) {
+        Offer offer = modelMapper.map(offerDto, Offer.class);
+
+        Offer savedOffer = offerRepository.save(offer);
+
+        return modelMapper.map(savedOffer, OfferDto.class);
+    }
 
-        Offer.OfferBuilder builder = Offer.builder();
-        Offer offer = builder
-                .name(offerDto.getName())
-                .owner(user)
-                .description(offerDto.getDescription())
-                .build();
+    public OfferDto updateOffer(long id, CreateOfferDto offerDto) {
+        Offer offer = offerRepository.getById(id);
+
+        modelMapper.map(offerDto, offer);
+
+        Offer updatedOffer = offerRepository.save(offer);
+
+        return modelMapper.map(updatedOffer, OfferDto.class);
+    }
+
+    public OfferDto deleteOffer(long id) {
+        Offer offerToDelete = offerRepository.getById(id);
```

Comment:
**question** I know it's a matter of personal taste but just wondering: Do you think that separating (almost) each line of code with empty line makes your method (and the whole class) easier to read/understand? Compare both formattings below:

```
    public OfferDto updateOffer(long id, CreateOfferDto offerDto) {
        Offer offer = offerRepository.getById(id);

        modelMapper.map(offerDto, offer);

        Offer updatedOffer = offerRepository.save(offer);

        return modelMapper.map(updatedOffer, OfferDto.class);
    }
```

```
    public OfferDto updateOffer(long id, CreateOfferDto offerDto) {
        Offer offer = offerRepository.getById(id);
        modelMapper.map(offerDto, offer);
        Offer updatedOffer = offerRepository.save(offer);
        return modelMapper.map(updatedOffer, OfferDto.class);
    }
```

Reply:
My general approach is: "If a method body is so complicated that you have to split its lines with empty lines, then it's a signal that you should extract some code to submethod. Otherwise empty lines should not be needed".

Reply:
You're right, spacing doesn't make any better. Will be fixed.